chessboard/schema.py

```
@@ -149,6 +159,194 @@ def coerce_dict(existing, changed):
+                 name=None):
+        """Extend voluptuous.Schema __init__ with `name`."""
+        self.name = name
+        super(DocumentedSchema, self).__init__(schema, required=required,
+                                               extra=extra)
+
+    def __call__(self, data):
+        """Validate data against this schema."""
+        try:
+            return super(DocumentedSchema, self).__call__(data)
+        except volup.MultipleInvalid as exc:
+            self.decorate_errors(exc.errors)
+            raise
+
+    def find_schema(self, error_path, schema=None):
+        """Find schema entry from error.path.
+
+        This function will navigate the schema based on the voluptuous error
+        path supplied. It will start from the current schema as a root (unless
+        an override is supplied in the `schema` parameter) and iterate down the
+        list of valid/allowed fields and return the last valid schema in the
+        path.
+
+        :returns: schema
+        """
+        if not schema:
+            schema = self
+
+        for node in error_path:
+            try:
+                if isinstance(schema, DocumentedSchema):
+                    schema = schema.get_field_schema(node)
+                elif isinstance(schema, DictOf):
+                    schema = schema.schema
+                elif isinstance(schema, list):
+                    schema = schema[0]
+                else:
+                    assert False, "Type '%s' not support." % type(schema)
+            except KeyError:
+                return schema
+        return schema
+
+    def decorate_errors(self, errors):
+        """Add `docs` key from schema docs to each error."""
+        for error in errors:
+            if hasattr(error, 'docs'):
+                continue
+            try:
+                definition = self.find_schema(error.path)
+                error.docs = DOCS[definition.name]
+            except KeyError:
```

When would a `KeyError` be raised? Why is it okay to swallow the error? This line has no test coverage.